packages/cryptoplease/lib/features/outgoing_tip_payments/src/bl/bloc.dart

```diff
@@ -0,0 +1,200 @@
+      txSent: (status) => _waitTx(status.tx, escrow: status.escrow),
+      txConfirmed: (status) => _createLink(
+        escrow: status.escrow,
+        token: payment.amount.token,
+      ),
+      linkReady: (status) async => status,
+      success: (status) async => status,
+      txFailure: (_) => _createTx(payment.amount),
+      txSendFailure: (status) => _sendTx(status.tx, escrow: status.escrow),
+      txWaitFailure: (status) => _waitTx(status.tx, escrow: status.escrow),
+      txLinksFailure: (status) => _createLink(
+        escrow: status.escrow,
+        token: payment.amount.token,
+      ),
+    );
+
+    await _repository.save(payment.copyWith(status: newStatus));
+
+    emit(state.remove(payment.id));
+
+    newStatus.map(
+      txCreated: (_) => add(OTEvent.process(payment.id)),
+      txSent: (_) => add(OTEvent.process(payment.id)),
+      txConfirmed: (_) => add(OTEvent.process(payment.id)),
+      linkReady: ignore,
+      success: ignore,
+      txFailure: ignore,
+      txSendFailure: ignore,
+      txWaitFailure: ignore,
+      txLinksFailure: ignore,
+    );
+  }
+
+  Future<OTStatus> _createTx(Amount amount) async {
+    try {
+      final escrowAccount = await Ed25519HDKeyPair.random();
+
+      final dto = CreatePaymentRequestDto(
+        senderAccount: _account.address,
+        escrowAccount: escrowAccount.address,
+        amount: amount.value,
+        cluster: apiCluster,
+      );
+
+      final tx = await _client
+          .createPayment(dto)
+          .then((it) => it.transaction)
+          .then(SignedTx.decode)
+          .then((it) => it.resign(_account));
+
+      return OTStatus.txCreated(tx, escrow: escrowAccount);
+    } on Exception {
+      return const OTStatus.txFailure();
+    }
+  }
+
+  Future<OTStatus> _sendTx(
+    SignedTx tx, {
+    required Ed25519HDKeyPair escrow,
+  }) async {
+    final result = await _txSender.send(tx);
+
+    return result.map(
+      sent: (_) => OTStatus.txSent(tx, escrow: escrow),
+      invalidBlockhash: (_) => const OTStatus.txFailure(),
+      failure: (it) => OTStatus.txFailure(reason: it.reason),
+      networkError: (_) => OTStatus.txSendFailure(tx, escrow: escrow),
+    );
+  }
+
+  Future<OTStatus> _waitTx(
+    SignedTx tx, {
+    required Ed25519HDKeyPair escrow,
+  }) async {
+    final result = await _txSender.wait(tx);
+
+    return result.map(
+      success: (_) => OTStatus.txConfirmed(escrow: escrow),
+      failure: (_) => const OTStatus.txFailure(),
+      networkError: (_) => OTStatus.txWaitFailure(tx, escrow: escrow),
+    );
+  }
+
+  Future<OTStatus> _createLink({
+    required Ed25519HDKeyPair escrow,
+    required Token token,
+  }) async {
+    final privateKey = await escrow.extract().then((value) => value.bytes.lock);
+    final key = base58encode(privateKey.toList());
+
+    final link = TipPaymentData(
+      key: key,
+      token: token.publicKey,
+    ).toUri();
```

Comment:
The link should be wrapped into Firebase deep link (and internal QR code scanner should be able to process it as well).